src/main/java/com/iexec/worker/feign/LoginService.java

```diff
@@ -60,24 +61,25 @@ public String login() {
         String workerAddress = credentialsService.getCredentials().getAddress();
         ECKeyPair ecKeyPair = credentialsService.getCredentials().getEcKeyPair();
 
-        ResponseEntity<String> challengeResponse = coreClient.getChallenge(workerAddress);
-        if (!challengeResponse.getStatusCode().is2xxSuccessful()) {
-            log.error("Cannot login, failed to get challenge [status:{}]", challengeResponse.getStatusCode());
-            return "";
+        String challenge = "";
+        try {
+            challenge = coreClient.getChallenge(workerAddress);
+        } catch (FeignException e) {
+            log.error("Cannot login, failed to get challenge [status:{}]", e.status());
```

Comment:
Shouldn't we return after this log? Otherwise, we'll have two similar logs "Cannot login, [...]" whereas the second one is useless/misleading.

Reply:
well spotted, see 8981ad7